app/src/main/java/org/simple/clinic/bp/entry/LinearLayoutWithPreImeKeyEventListener.kt

```diff
@@ -0,0 +1,20 @@
+package org.simple.clinic.bp.entry
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.KeyEvent
+import android.widget.LinearLayout
+
+class LinearLayoutWithPreImeKeyEventListener(context: Context, attrs: AttributeSet) : LinearLayout(context, attrs) {
+
+  lateinit var backKeyPressInterceptor: () -> Unit
```

Comment:
It seems like `lateinit` is being used wrongly here since it is meant to be for something that **will** be initialized at some point in the future, but we are depending on an external class to set it **if** they need to.

Reply:
TODO: throw an error instead of using `lateinit`.